plugins/org.python.pydev.parser/src/org/python/pydev/parser/grammar310/TreeBuilder310.java

```diff
@@ -831,19 +745,94 @@ public final SimpleNode onCloseNode(SimpleNode n, int arity) throws Exception {
                             patterns[i] = (patternType) popNode;
                         } else if (popNode instanceof exprType) {
                             patterns[i] = new MatchValue((exprType) popNode);
+                        } else {
+                            patterns[i] = getDefaultInvalidPattern();
                         }
                     }
                     return new MatchSequence(patterns, 0);
                 }
-                Log.log("Expected arity to be 1 or > 1 here.");
-                return null;
+                addAndReportException(SimpleNode.class.getName());
+                return new MatchSequence(new patternType[0], 0);
 
             default:
                 Log.log(("Error at TreeBuilder: default not treated:" + n.getId()));
                 return null;
         }
     }
 
+    private match_caseType getDefaultInvalidMatchCase() {
+        return new match_caseType(getDefaultInvalidPattern(), null, getDefaultBody());
+    }
+
+    private stmtType[] getDefaultBody() {
+        return new stmtType[] { new Pass() };
+    }
+
+    private patternType getDefaultInvalidPattern() {
+        return new MatchValue(getDefaultInvalidExpr());
+    }
+
+    private exprType getDefaultInvalidExpr() {
+        return new Name("$INVALID$", Name.Artificial, false);
+    }
+
+    private void popSurplus(int arity, int max) {
+        if (arity > max) {
+            Log.log("Popping all surplus nodes. Expected arity: " + max + ". Actual: " + arity);
```

Comment:
This should actually be `addAndReportException` (you can bind it to the first node popped).